### PR TITLE
[code-infra] Latest vitest picker changes

### PR DESCRIPTION
**Pull request by JCQuintas**
Same as with previous PRs, goal is to trim down changes in the [main PR](https://github.com/mui/mui-x/pull/14508)

### CODE REVIEW

File: test/utils/pickers/describeRangeValidation/testDayViewRangeValidation.tsx [hunk 1]
```diff
@@ -67,84 +66,74 @@ export const testDayViewRangeValidation: DescribeRangeValidationTestSuite = (
         testDisabledDate('11', [true, true], !isDesktop || includesTimeView);
       });
 
-      describe('with fake timers', () => {
-        // TODO: temporary for vitest. Can move to `vi.useFakeTimers`
-        let timer: SinonFakeTimers | null = null;
-        beforeEach(() => {
-          timer = useFakeTimers({ now: new Date(2018, 0, 1), toFake: ['Date'] });
```
Comment (LukasTy) [inline test/utils/pickers/describeRangeValidation/testDayViewRangeValidation.tsx:74]: Haven't we already tried removing this time mocking and later encountered edge-cases at certain times? 🤔

Comment (JCQuintas) [reply test/utils/pickers/describeRangeValidation/testDayViewRangeValidation.tsx:74]: I can't remember 🤔

I can also keep it in for this describe block if you think it is best

Comment (LukasTy) [reply test/utils/pickers/describeRangeValidation/testDayViewRangeValidation.tsx:74]: I prefer keeping it here. 🤔

Comment (JCQuintas) [reply test/utils/pickers/describeRangeValidation/testDayViewRangeValidation.tsx]: reverted

**Review by LukasTy — APPROVED** [review]
